docs/en/usage/java-sdk-user-guide.md

```diff
@@ -1 +1,1254 @@
+
+### 3.1.2 Listening for configuration change events
+
+Listening for configuration change events is only used when the application really cares about configuration changes and needs to be notified when the configuration changes, e.g. when the database connection string changes and the connection needs to be rebuilt, etc.
+
+If you just want to fetch the latest configuration every time, just follow the example above and call **config.getProperty**.
+
+```java
+Config config = ConfigService.getAppConfig(); //config instance is singleton for each namespace and is never null
+config.addChangeListener(new ConfigChangeListener() {
+    @Override
+    public void onChange(ConfigChangeEvent changeEvent) {
+        System.out.println("Changes for namespace " + changeEvent.getNamespace());
+        for (String key : changeEvent.changedKeys()) {
+            ConfigChange change = changeEvent.getChange(key);
+            System.out.println(String.format("Found change - key: %s, oldValue: %s, newValue: %s, changeType: %s", change.getPropertyName(), change. getOldValue(), change.getNewValue(), change.getChangeType()));
+        }
+    }
+});
+```
+
+### 3.1.3 Get the configuration of the public Namespace
+
+``` java
+String somePublicNamespace = "CAT";
+Config config = ConfigService.getConfig(somePublicNamespace); //config instance is singleton for each namespace and is never null
+String someKey = "someKeyFromPublicNamespace";
+String someDefaultValue = "someDefaultValueForTheKey";
+String value = config.getProperty(someKey, someDefaultValue);
+```
+
+### 3.1.4 Get the configuration of a non-properties format namespace
+
+#### 3.1.4.1 Namespace in yaml/yml format
+
+Apollo-Client version 1.3.0 starts to make better support for `yaml/yml`, which is used in the same way as properties format.
+
+```java
+Config config = ConfigService.getConfig("application.yml");
+String someKey = "someKeyFromYmlNamespace";
+String someDefaultValue = "someDefaultValueForTheKey";
+String value = config.getProperty(someKey, someDefaultValue);
+```
+
+#### 3.1.4.2 Namespace in non-yaml/yml format
+
+To get it, you need to use the `ConfigService.getConfigFile` interface and specify the Format, such as `ConfigFileFormat.XML`.
+
+```java
+String someNamespace = "test";
+ConfigFile configFile = ConfigService.getConfigFile("test", ConfigFileFormat.XML);
+String content = configFile.getContent();
+```
+
+## 3.2 Spring integration approach
+
+### 3.2.1 Configuration
+
+Apollo also supports integration with Spring (Spring 3.1.1+), and only requires some simple configuration.
+
+Apollo currently supports both the more traditional `XML-based` configuration and the currently more popular `Java-based (recommended)` configuration.
+
+In case of Spring Boot environments, it is recommended to refer to [3.2.1.3 Spring Boot integration methods (recommended)](#_3213-Spring-Boot-integration-methods-(recommended)) for configuration.
```

Comment:
```suggestion
In case of Spring Boot environments, it is recommended to refer to [3.2.1.3 Spring Boot integration methods (recommended)](#_3213-spring-boot-integration-methods-recommended) for configuration.
```